test/compress/issue-1466.js

```diff
@@ -0,0 +1,56 @@
+same_variable_in_multiple_for_loop: {
+    options = {}
+	mangle = {}
+    input: { 
+		for (let i = 0; i < 3; i++) {
+			let a = 100;
+			console.log(i + ",a=" + a);
+			for (let i = 0; i < 2; i++) {
+				console.log(i + " ==== " + a);
+				let c = 2;
+				console.log(c);
+			}
+		}
+	}
+    expect: {
+		for(let o=0;o<3;o++){let l=100;console.log(o+",a="+l);for(let o=0;o<2;o++){console.log(o+" ==== "+l);let c=2;console.log(c)}}
+	}
+}
+same_variable_in_multiple_forOf: {
+    options = {}
+	mangle = {}
+    input: { 
+		var test = ["a","b","c"];
+		for (let tmp of test) {
+			console.log(tmp);
+			let dd ;
+			dd = ["e","f","g"];
+			for (let tmp of dd) {
+				console.log(tmp);
+			}
+		}
+	}
+    expect: {
+		var test=["a","b","c"];for(let o of test){console.log(o);let e;e=["e","f","g"];for(let o of e)console.log(o)}
+	}
+}
+
+same_variable_in_multiple_forIn: {
+    options = {}
+	mangle = {}
+    input: { 
+		var test = ["a","b","c"];
+		for (let tmp in test) {
+			console.log("for in===="+tmp);
+			let dd ;
+			dd= ["e","f","g"];
+			for (let tmp in test) {
+				console.log(tmp);
+			}
+		}
+	}
+    expect: {
+		var test=["a","b","c"];for(let o in test){console.log("for in===="+o);let e;e=["e","f","g"];for(let o in test)console.log(o)}
```

Comment:
@alexlamsl 
the cases failed by run npm test if  i omit  changes in lib/scope.js.
I will soon format the code.